cmd/support-upload.go

```diff
@@ -0,0 +1,146 @@
+	"github.com/minio/pkg/v2/console"
+)
+
+// profile command flags.
+var (
+	uploadFlags = append(globalFlags,
+		cli.IntFlag{
+			Name:  "issue",
+			Usage: "SUBNET issue number to which the file is to be uploaded",
+		},
+		cli.StringFlag{
+			Name:  "comment",
+			Usage: "comment to be posted on the issue along with the file",
+		},
+		cli.BoolFlag{
+			Name:   "dev",
+			Usage:  "Development mode",
+			Hidden: true,
+		},
+	)
+)
+
+type supportUploadMessage struct {
+	Status   string `json:"status"`
+	IssueNum int    `json:"-"`
+	IssueURL string `json:"issueUrl"`
+}
+
+// String colorized upload message
+func (s supportUploadMessage) String() string {
+	msg := fmt.Sprintf("File uploaded to SUBNET successfully. Click here to visit the issue: %s", subnetIssueURL(s.IssueNum))
+	return console.Colorize(supportSuccessMsgTag, msg)
+}
+
+// JSON jsonified upload message
+func (s supportUploadMessage) JSON() string {
+	return toJSON(s)
+}
+
+var supportUploadCmd = cli.Command{
+	Name:            "upload",
+	Usage:           "upload file to SUBNET issue",
+	Action:          mainSupportUpload,
+	OnUsageError:    onUsageError,
+	Before:          setGlobalsFromContext,
+	Flags:           uploadFlags,
+	HideHelpCommand: true,
+	CustomHelpTemplate: `NAME:
+  {{.HelpName}} - {{.Usage}}
+
+USAGE:
+  {{.HelpName}} [FLAGS] ALIAS FILE
+
+FLAGS:
+  {{range .VisibleFlags}}{{.}}
+  {{end}}
+EXAMPLES:
+  1. Upload file './trace.log' for cluster 'myminio' to SUBNET issue number 10
+     {{.Prompt}} {{.HelpName}} --issue 10 myminio ./trace.log
+
+  2. Upload file './trace.log' for cluster 'myminio' to SUBNET issue number 10 with comment 'here is the trace log'
+     {{.Prompt}} {{.HelpName}} --issue 10 --comment "here is the trace log" myminio ./trace.log 
+`,
+}
+
+func checkSupportUploadSyntax(ctx *cli.Context) {
+	if len(ctx.Args()) != 2 {
+		showCommandHelpAndExit(ctx, 1) // last argument is exit code
+	}
+
+	if ctx.Int("issue") <= 0 {
+		fatal(errDummy().Trace(), "Invalid issue number")
+	}
+}
+
+// mainSupportUpload is the handle for "mc support upload" command.
+func mainSupportUpload(ctx *cli.Context) error {
+	// Check for command syntax
+	checkSupportUploadSyntax(ctx)
+	setSuccessMessageColor()
+
+	// Get the alias parameter from cli
+	aliasedURL := ctx.Args().Get(0)
+	alias, apiKey := initSubnetConnectivity(ctx, aliasedURL, true, true)
+	if len(apiKey) == 0 {
+		// api key not passed as flag. Check that the cluster is registered.
+		apiKey = validateClusterRegistered(alias, true)
+	}
+
+	// Main execution
+	execSupportUpload(ctx, alias, apiKey)
+	return nil
+}
+
+func execSupportUpload(ctx *cli.Context, alias, apiKey string) {
+	filePath := ctx.Args().Get(1)
+	issueNum := ctx.Int("issue")
+	msg := ctx.String("comment")
+
+	f, e := os.Open(filePath)
+	fatalIf(probe.NewError(e), "Unable to open file '%s'", filePath)
+	defer f.Close()
+	fi, e := f.Stat()
+	fatalIf(probe.NewError(e), "Unable to get file info for '%s'", filePath)
+	filename := fi.Name()
```

Comment:
L126 to L131 is not needed in my opinion. You are not using f later,  filename is just the name of the file and not its path. I think this will not work if specify a full path of a file

Reply:
`filename` is used for creation of upload URL only and while upload we are using `filePath` at L141, so works. tested with `./mc support upload --issue 62 --comment "file txt" test-shireesh /home/shubhendu/Downloads/after-stop.txt --dev` and works as expected.

Reply:
okay, it just looks an overkill. We can use os.Stat() and filepath.Base()